packages/Algernon.package/ALGWorldMenuType.class/class/menuEntries.st

```diff
@@ -1,16 +1,9 @@
 algertem lists
 menuEntries
-	|entries|
-	
-	entries := OrderedCollection new.	
 
-	((self menus
-		collect: [ :each | ALGNamedWorldMenuEntry newFrom: each subMenu name: each contents ])
-		select: [ :each | each isMenu ])
-		do: [ :each |
-			entries addAll: (
-				(self extractMenuEntriesFrom: each)
-					collect: [ :entry | ALGNamedWorldMenuEntry newFrom: entry name: each name]) ].
-	
-	^ entries select: [:each | each morph isKindOf: MenuItemMorph]
-	
\ No newline at end of file
+	^ ((((self dockingBar submorphs
+		collect: [:morph | ALGNamedWorldMenuEntry newFrom: morph])
+		select: [:entry | entry isValid])
+		select: [:entry | entry hasEntries])
+		select: [:entry | entry name ~= 'Windows' ])
+		collect: [:entry | entry category: entry name]
```

Comment:
last `collect` should be a `do`, right? Maybe something like
```smalltalk
^ (self make
          complicated
          stuff)
        do: [ ... ];
        yourself
```

better describes what is going on here? Or go with a temporary variable